source/observables.txt

```diff
@@ -0,0 +1,243 @@
+      (restaurant_id,BsonString{value='41485121'}))
+      Iterable((_id,...), ..., (name,BsonString{value='Hospoda'}),
+      (restaurant_id,BsonString{value='41569184'}))
+      Iterable((_id,...), ..., (name,BsonString{value='Olde Prague Tavern'}),
+      (restaurant_id,BsonString{value='41711983'}))
+      Processed all results
+
+Access Write Operation Results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To access data retrieved by a write operation, create an ``Observable[T]``
+to store the operation results. Then, subscribe to the observable and
+override the ``Observer`` class callback methods to process the results.
+
+This example inserts documents into the ``restaurants`` database
+in which the ``cuisine`` value is ``"Nepalese"``. To retrieve
+and process results, the example assigns an ``Observable[InsertManyResult]``
+to the operation and performs the following actions:
+
+- Calls the ``subscribe()`` method to subscribe to the ``Observable``
+  and passes an ``Observer`` as a parameter
+- Customizes the ``onNext()`` method to print the result of the insert
+  operation, returned as an ``InsertManyResult``
+- Customizes the ``onError()`` method to print any errors
+- Customizes the ``onComplete()`` methods to print a message after all
+  the results from the ``Observable`` are processed
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/observables.scala
+      :language: scala
+      :start-after: start-read-observable
+      :end-before: end-read-observable
+      :dedent:
+
+   .. output::
+      :language: console
+      :visible: false
+
+      AcknowledgedInsertManyResult{insertedIds={0=BsonObjectId{value=...},
+      1=BsonObjectId{value=...}}}
+      Processed all results
+
+Use Lambda Functions to Process Results
+---------------------------------------
+
+Instead of explicitly overriding the callback functions from the
+``Observer`` class, you can use lambda functions to concisely
+process operation results. These functions allow you to use the
+``=>`` arrow notation to customize the implementation of ``onNext()``,
+``onError()``, and ``onComplete()``.
+
+.. tip::
+
+   To learn more about lambda, or anonymous, functions,
+   see the :wikipedia:`Anonymous function <w/index.php?title=Anonymous_function&oldid=1252038929>`
+   Wikipedia entry.
+
+Example
+~~~~~~~
+
+This example queries the ``restaurants`` database for each
+distinct value of the ``borough`` field. The code subscribes
+to the ``Observable`` returned by the ``distinct()`` method, then
+uses lambda functions to print results and handle errors:
+
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/observables.scala
+      :language: scala
+      :start-after: start-lambda
+      :end-before: end-lambda
+      :dedent:
+
+   .. output::
+      :language: console
+      :visible: false
+
+      Bronx
+      Brooklyn
+      Manhattan
+      Missing
+      Queens
+      Staten Island
+      Processed all results
+
```

Comment:
Suggestion: I think we should also mention the `Await.result()` method as an alternative to the callbacks/lambdas mentioned on this page. See [Time Series](https://deploy-preview-86--docs-scala.netlify.app/databases-collections/time-series/) for examples, as this would probably warrant its own section on this page.

Reply:
Added a section about futures!